pkg/virt-controller/watch/vm.go

```diff
@@ -179,20 +181,49 @@ func NewVMController(vmiInformer cache.SharedIndexInformer,
 	return c, nil
 }
 
-type sarProxy struct {
-	client kubecli.KubevirtClient
+type authProxy struct {
+	client             kubecli.KubevirtClient
+	dataSourceInformer cache.SharedIndexInformer
+	namespaceStore     cache.Store
 }
 
-func (p *sarProxy) Create(sar *authv1.SubjectAccessReview) (*authv1.SubjectAccessReview, error) {
+func (p *authProxy) CreateSar(sar *authv1.SubjectAccessReview) (*authv1.SubjectAccessReview, error) {
 	return p.client.AuthorizationV1().SubjectAccessReviews().Create(context.Background(), sar, v1.CreateOptions{})
 }
 
+func (p *authProxy) GetNamespace(name string) (*k8score.Namespace, error) {
+	obj, exists, err := p.namespaceStore.GetByKey(name)
+	if err != nil {
+		return nil, err
+	} else if !exists {
+		return nil, fmt.Errorf("namespace %s does not exist", name)
+	}
+
+	ns := obj.(*k8score.Namespace)
```

Comment:
When dealing with informers, I think it is good practice to return/work with a deep copy of the object in the cache.

Reply:
makes sense, done